payments/models.py

```diff
@@ -587,6 +585,9 @@ def clean(self):
             # if due_date change is allowed
             self._check_due_date(old_instance)
 
+        if self.status == OrderStatus.OFFERED and not self.due_date:
```

Comment:
Do we want to allow Orders to not have due_date if they are in ERROR? The only time an order is set to error state is when send the invoice fails, so it should have a due date at the time?

Reply:
I don't think the Orders would need a due_date in the case of DRAFTED -> ERROR state, but they would get the due_date when the order is set from ERROR -> OFFERED.

Reply:
Alrighty. So do I remove ` and not self.due_date` so that it gets a new due date even though it has one?

Reply:
Or perhaps the best solution would be to clean the due date in `leases.services.invoice.BaseInvoicingService.fail_order`

Reply:
Sounds good 👍